src/java.base/share/classes/jdk/internal/classfile/impl/verifier/ParserVerifier.java

```diff
@@ -240,90 +244,104 @@ private void verifyAttribute(AttributedElement ae, Attribute<?> a, List<VerifyEr
+                size = 1 + 4 * mpa.parameters().size();
+            case NestHostAttribute nha -> {
+                nha.nestHost();
+                size = 2;
             }
-            case LineNumberTableAttribute lta -> 2 + 4 * lta.lineNumbers().size();
-            case LocalVariableTableAttribute lvta -> 2 + 10 * lvta.localVariables().size();
-            case LocalVariableTypeTableAttribute lvta -> 2 + 10 * lvta.localVariableTypes().size();
-            case MethodParametersAttribute mpa -> 1 + 4 * mpa.parameters().size();
-            case NestHostAttribute nha -> 2;
             case NestMembersAttribute nma -> {
                 if (ae.findAttribute(Attributes.NEST_HOST).isPresent()) {
                     errors.add(new VerifyError("Conflicting NestHost and NestMembers attributes in %s".formatted(toString(ae))));
                 }
-                yield 2 + 2 * nma.nestMembers().size();
+                size = 2 + 2 * nma.nestMembers().size();
             }
             case PermittedSubclassesAttribute psa -> {
                 if (classModel.flags().has(AccessFlag.FINAL)) {
                     errors.add(new VerifyError("PermittedSubclasses attribute in final %s".formatted(toString(ae))));
                 }
-                yield 2 + 2 * psa.permittedSubclasses().size();
+                size = 2 + 2 * psa.permittedSubclasses().size();
             }
             case RecordAttribute ra -> {
-                int l = 2;
+                size = 2;
                 for (var rc : ra.components()) {
-                    l += 6;
+                    size += 6;
                     for (var rca : rc.attributes()) {
-                        l += 6 + ((BoundAttribute)rca).payloadLen();
+                        size += 6 + ((BoundAttribute)rca).payloadLen();
                     }
                 }
-                yield l;
-            }
-            case RuntimeVisibleAnnotationsAttribute aa -> {
-                int l = 2;
-                for (var an : aa.annotations()) {
-                    l += annotationSize(an);
-                }
-                yield l;
-            }
-            case RuntimeInvisibleAnnotationsAttribute aa -> {
-                int l = 2;
-                for (var an : aa.annotations()) {
-                    l += annotationSize(an);
-                }
-                yield l;
             }
+            case RuntimeVisibleAnnotationsAttribute aa ->
+                size = annotationsSize(aa.annotations());
+            case RuntimeInvisibleAnnotationsAttribute aa ->
+                size = annotationsSize(aa.annotations());
             case RuntimeVisibleParameterAnnotationsAttribute aa -> {
-                int l = 1;
+                size = 1;
                 for (var ans : aa.parameterAnnotations()) {
-                    l += 2;
-                    for (var an : ans) {
-                        l += annotationSize(an);
-                    }
+                    size += annotationsSize(ans);
```

Comment:
Shouldn't we call the new method here (passing `parameterAnnotations`) ?

Reply:
Yes, I've compacted it a bit more, thanks.